### PR TITLE
feat: document custom 500

### PR DESCRIPTION
**Pull request by florian-lefebvre**
#### Description (required)

Documents the usage of custom 500 error pages.

#### Related issues & labels (optional)

Code PR https://github.com/withastro/astro/pull/11244

### CODE REVIEW

**Comment by sarah11918** [comment]
This looks good to me! There was a problem with the deployment build, so I've updated branch so that it will run again. When checks pass, I'll approve!

**Review by sarah11918 — APPROVED** [review]
Looks good, thank you!

File: src/content/docs/en/basics/astro-pages.mdx [hunk 1]
```diff
@@ -96,10 +96,18 @@ Files with the `.html` file extension can be placed in the `src/pages/` director
...
 
 This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
 
+## Custom 500 Error Page
+
+For a custom 500 error page to show for pages in your `hybrid` or `server` project that are [rendered on demand](/en/guides/server-side-rendering/#enable-on-demand-server-rendering), create the file `src/pages/500.astro`. This custom page is not available for prerendered pages.
+
+If an error occurs rendering this page, your host's default 500 error page will be shown to your visitor.
+
+During development, if you have a `500.astro`, the error thrown at runtime is logged in your terminal, as opposed to being shown in the error overlay.
```
Comment (florian-lefebvre) [inline src/content/docs/en/basics/astro-pages.mdx:109]: Should we specify since what version this happens?

Comment (sarah11918) [reply src/content/docs/en/basics/astro-pages.mdx:109]: Good call. Throw in a `<Since>` component between those two paragraphs! Check the deploy preview to make sure it looks OK, then fine to merge!

Comment (florian-lefebvre) [reply src/content/docs/en/basics/astro-pages.mdx:109]:
```suggestion
<p><Since v="4.10.3" /></p>

During development, if you have a `500.astro`, the error thrown at runtime is logged in your terminal, as opposed to being shown in the error overlay.
```
Why do we need to wrap it inside a <p>?

Comment (sarah11918) [reply src/content/docs/en/basics/astro-pages.mdx:109]: If you can get it to look right without one, go ahead! That's the pattern I use for spacing!

**Comment by sarah11918** [comment]
I am leaving the computer, so this one's on you to merge when you're happy!

**Comment by florian-lefebvre** [comment]
Alright thanks! The patch is planned for Monday I believe

File: src/content/docs/en/basics/astro-pages.mdx [hunk 2]
```diff
@@ -96,10 +96,20 @@ Files with the `.html` file extension can be placed in the `src/pages/` director
...
 ## Custom 404 Error Page
 
-For a custom 404 error page, you can create a `404.astro` or `404.md` file in `/src/pages`.
+For a custom 404 error page, you can create a `404.astro` or `404.md` file in `src/pages`.
 
 This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
 
+## Custom 500 Error Page
+
+For a custom 500 error page to show for pages in your `hybrid` or `server` project that are [rendered on demand](/en/guides/server-side-rendering/#enable-on-demand-server-rendering), create the file `src/pages/500.astro`. This custom page is not available for prerendered pages.
```
Comment (florian-lefebvre) [inline src/content/docs/en/basics/astro-pages.mdx:105]:
```suggestion
For a custom 500 error page to show for pages [rendered on demand](/en/guides/server-side-rendering/#enable-on-demand-server-rendering), create the file `src/pages/500.astro`. This custom page is not available for prerendered pages and can't be prerendered.
```
@sarah11918 wdyt?

Comment (sarah11918) [reply src/content/docs/en/basics/astro-pages.mdx:105]: I'm in favour of the shorter wording! I would probably keep "that" as it helps with readability:

> for pages that are rendered on demand

It's just less dense and can be easier to read. But, LGTM! @florian-lefebvre
